src/index.js

```diff
@@ -4,21 +4,28 @@ import "./index.css";
 import App from "./App";
 import { makeServer } from "./server";
 import { BrowserRouter as Router } from "react-router-dom";
-import { AuthProvider, CartProvider, FilterProvider } from "./context";
+import {
+  AuthProvider,
+  CartProvider,
+  FilterProvider,
+  WishlistProvider,
+} from "./context";
 
 // Call make Server
 makeServer();
 
 ReactDOM.render(
   <React.StrictMode>
     <Router>
```

Comment:
You should have your `Router` as the innermost provider so that it has access to all other providers and their data.

Reply:
I was thinking the reverse. Doesn't making it the outermost give it access to data of all other providers?

Reply:
No, if the component doesn't find the provider for a state which it uses, it looks upward hence, suppose you're router is using some state of the cart then to look for cart provider it would search it above itself in the dom and not inside.

Reply:
Oh okay, I get it now! Thanks for clearing my confusion  😁